config/watcher.go

```diff
@@ -66,11 +69,25 @@ func newCloudWatcher(ctx context.Context, config *Config, logger logging.Logger)
 			if time.Now().After(nextCheckForNewCert) {
 				checkForNewCert = true
 			}
-			newConfig, err := readFromCloud(cancelCtx, config, prevCfg, false, checkForNewCert, logger)
+
+			if conn == nil {
+				var err error
+				conn, err = CreateNewGRPCClient(ctx, config.Cloud, logger)
+				if err != nil {
+					conn = nil
+					logger.Warn("failed to connect to app")
+					continue
+				}
+			}
+			rr := remoteReader{apppb.NewRobotServiceClient(conn)}
```

Comment:
i think we should use `newRemoteReader` here too, even if that means connection management goes into the remoteReader itself

Reply:
ah yeah, not sure why I didn't use that...